src/components/KCopy/KCopy.vue

```diff
@@ -0,0 +1,274 @@
+<template>
+  <div
+    v-if="!isBadge"
+    class="k-copy"
+    :class="truncate ? null : 'copy-truncated'"
+  >
+    <KTooltip
+      v-if="format !== 'hidden'"
+      :class="truncate ? null : 'copy-content'"
+      data-testid="copy-content"
+      :label="contentTooltip"
+      placement="bottomStart"
+      position-fixed
+    >
+      <div
+        :class="[
+          'content-container',
+          truncate ? 'truncated-content' : 'non-truncated-content',
+          monospace ? 'monospace' : null
+        ]"
+        :style="widthStyle"
+      >
+        {{ contentFormat }}
+      </div>
+    </KTooltip>
+
+    <KTooltip
+      class="content-icon-wrapper"
+      :label="`${props.badgeText}` ? iconTitle : tooltipText"
+      max-width="500px"
+      placement="bottomStart"
+      position-fixed
+    >
+      <KClipboardProvider v-slot="{ copyToClipboard }">
+        <span
+          data-testid="copy-to-clipboard"
+          role="button"
+          tabindex="0"
+          @click.stop="copyIdToClipboard(copyToClipboard)"
+        >
+          <KIcon
+            class="content-icon"
+            :hide-title="!!copyTooltip || undefined"
+            icon="copy"
+            size="16"
+            :title="iconTitle"
+          />
+        </span>
+      </KClipboardProvider>
+    </KTooltip>
+  </div>
+
+  <div
+    v-else
+    class="k-copy-badge"
```

Comment:
issue: the wrapper container should be the only element with a `k-*` class.

This component has two top-level containers; `k-copy` and `k-copy-badge` -- it should just be `k-copy` and then modifier classes added as needed for the badge format instead of an entirely new wrapper